occurrences/tests/test_api.py

```diff
@@ -1006,6 +1007,48 @@ def test_enrol_occurrence(snapshot, api_client, mock_get_event_data):
     snapshot.assert_match(executed)
 
 
+def test_enrol_occurrence_with_captcha(
+    snapshot, api_client, mock_get_event_data, settings, mock_recaptcha_data
+):
+    settings.CAPTCHA_ENABLED = True
+    study_group_15 = StudyGroupFactory(group_size=15)
+    # Current date froze on 2020-01-04:
+    p_event_1 = PalvelutarjotinEventFactory(
+        enrolment_start=datetime(2020, 1, 3, 0, 0, 0, tzinfo=timezone.now().tzinfo),
+        enrolment_end_days=2,
+    )
+    occurrence = OccurrenceFactory(
+        start_time=datetime(2020, 1, 6, 0, 0, 0, tzinfo=timezone.now().tzinfo),
+        p_event=p_event_1,
+        min_group_size=10,
+        max_group_size=20,
+        amount_of_seats=50,
+    )
+
+    variables = {
+        "input": {
+            "occurrenceIds": [to_global_id("OccurrenceNode", occurrence.id)],
+            "studyGroup": {
+                "person": {
+                    "id": to_global_id("PersonNode", study_group_15.person.id),
+                    "name": study_group_15.person.name,
+                    "emailAddress": study_group_15.person.email_address,
+                },
+                "name": "To be created group",
+                "groupSize": study_group_15.group_size,
+                "groupName": study_group_15.group_name,
+                "studyLevel": study_group_15.study_level.upper(),
+                "amountOfAdult": study_group_15.amount_of_adult,
+            },
+        }
+    }
+    executed = api_client.execute(ENROL_OCCURRENCE_MUTATION, variables=variables)
+    assert_match_error_code(executed, CAPTCHA_VALIDATION_FAILED_ERROR)
```

Comment:
Hmm, shouldn't this error actually be a GraphQL one saying `captchaKey` is a required field that is missing?

Reply:
Putting the setting as part of schema parameter make it quite difficult to run during tests because Scheme is built before the settings fixture is injected in the test case. In this case, the `captchaKey` wasn't required in the schema even though that setting was set to True at the beginning of the test. I removed the `required=settings.CAPTCHA_ENABLED` and make it always optional in the mutation so it'd be easier to control the input during test.

Reply:
Right. I think it might have been possible to get it working using Django's [override_settings](https://docs.djangoproject.com/en/3.1/topics/testing/tools/#django.test.override_settings). Anyway, the change you made is of course fine too.